src/utils/uniq.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns the input array without duplicates values.
+ * All values are unique.
+ * @param {Array.<*>} arr
+ * @returns {Array.<*>}
+ */
+export default function uniq<T>(arr: T[]) {
+  if (typeof (window as any).Set === "function") {
+    return Array.from(new Set(arr));
+  } else {
+    return arr.filter((val, i, self) => self.indexOf(val) === i);
+  }
+}
```

Comment:
This means that this check will be performed each time `uniq` is called.

Why not more something like:

```ts
export default typeof (window as any).Set === "function" ?
  function uniqFromSet(arr) {
    return Array.from(new Set(arr));
  } :
  function uniqFromFilter(arr) {
    return arr.filter((val, i, self) => self.indexOf(val) === i);
  };
```